File: typescript/resistor-color-duo/resistor-color-duo.ts
const colors: { [c: string]: number } = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
};

export function decodedValue([c1, c2]: string[]): number {
  return 10 * colors[c1] + colors[c2];
}

File: python/resistor-color-duo/resistor_color_duo.py
"""Two-coloured resistor value
"""


def value(colors):
    return 10 * color_value.index(colors[0]) + color_value.index(colors[1])


color_value = ["black", "brown", "red", "orange",
               "yellow", "green", "blue", "violet", "grey", "white"]
